extensions/PhabBugz/lib/WebService.pm

```diff
@@ -398,8 +399,64 @@ sub _phabricator_precheck {
         unless $user->login eq PHAB_AUTOMATION_USER;
 }
 
+sub set_build_target {
+    my ( $self, $params ) = @_;
+
+    # Phabricator only supports sending credentials via HTTP Basic Auth
+    # so we exploit that function to pass in an API key as the password
+    # of basic auth. BMO does not support basic auth but does support
+    # use of API keys.
+    my $http_auth = Bugzilla->cgi->http('Authorization');
+    $http_auth =~ s/^Basic\s+//;
+    $http_auth = decode_base64($http_auth);
+    my ($login, $api_key) = split(':', $http_auth);
+    $params->{'Bugzilla_login'}   = $login;
+    $params->{'Bugzilla_api_key'} = $api_key;
+
```

Comment:
I thought we were already doing this higher in the auth stack?

Reply:
Not that I can find. We used to do it for another PhabBugz API method revision() but that is no longer being used.